packages/domain/lib/usecase/get_movie_list_usecase.dart

```diff
@@ -1,37 +1,98 @@
+import 'package:domain/model/data_model.dart';
+import 'package:domain/model/cache_models/movie_cache_model.dart';
 import '../enum/movie_type.dart';
+import '../repository/local_storage_repository.dart';
 import '../utils/const.dart';
-import '../model/movie/movie_response_model.dart';
 import '../repository/trakt_api_repository.dart';
 import 'usecase.dart';
 
 class GetMovieListUseCase
-    implements UseCaseParams<MovieType, Future<List<MovieResponse>>> {
+    implements UseCaseParams<MovieType, Future<List<MovieCache>>> {
   final TraktAPIRepository _repository;
+  final LocalStorageRepository _localStorageRepository;
 
-  GetMovieListUseCase(this._repository);
+  GetMovieListUseCase(
+    this._repository,
+    this._localStorageRepository,
+  );
 
   @override
-  Future<List<MovieResponse>> call(MovieType movieType) async {
-    final List<MovieResponse> jsonMovies = [];
-    final response = movieType == MovieType.anticipated
-        ? await _repository.getMovieAnticipatedData()
-        : await _repository.getMovieTrendingData();
-    final pageCount = int.parse(response.headers[Config.pageCount][0]);
-    final itemCount =
-        pageCount >= 5 ? 50 : int.parse(response.headers[Config.itemCount][0]);
-    await _getMovies(itemCount, jsonMovies, movieType);
-    return jsonMovies;
+  Future<List<MovieCache>> call(MovieType currentMovieType) async {
+    final List<MovieCache> jsonMovies = [];
+    final List<int> movieCacheIds = [];
+    if (await _isTimestampNewDay(
```

Comment:
method with such name should compare only 2 dates, no async work. rename it or change the logic